components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -2007,14 +2010,38 @@ private void setSpOIDCProperties(Map<String, List<String>> spOIDCProperties, OAu
                     Boolean.parseBoolean(isAccessTokenClaimsSeparationEnabled));
         }
 
-        boolean hybridFlowEnabled = Boolean.parseBoolean(getFirstPropertyValue(spOIDCProperties,
-                HYBRID_FLOW_ENABLED));
-        oauthApp.setHybridFlowEnabled(hybridFlowEnabled);
+        String hybridFlowEnabledProperty = getFirstPropertyValue(spOIDCProperties, HYBRID_FLOW_ENABLED);
+        if (hybridFlowEnabledProperty == null) {
+            // This application doesn't have hybridFlowEnabled property, hence providing the previous behaviour.
+            // which is enabling the hybrid flow with all allowed response types.
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(String.format("This application with consumer key %s doesn't have hybridFlowEnabled " +
+                        "property, hence providing the previous behaviour, which is enabling the hybrid flow with " +
+                        "all configured response types.", oauthApp.getOauthConsumerKey()));
+            }
+            switch (configuredHybridResponseTypes.size()) {
+                case 0:
+                    oauthApp.setHybridFlowEnabled(false);
+                    oauthApp.setHybridFlowResponseType(null);
+                    break;
+                case 1:
+                    oauthApp.setHybridFlowEnabled(true);
+                    oauthApp.setHybridFlowResponseType(configuredHybridResponseTypes.get(0));
+                    break;
+                default:
```

Comment:
Here when the user assigned two hybrid flow response types, it will enable all the hybrid flow responses. IMO it's incorrect, we need to improve it. 

If this will be addressed with a separate effort, +1 to proceed with the current approach.